software/speller/data_collection_platform/backend/dcp/api.py

```diff
@@ -151,6 +152,10 @@ def openbci_stop(process_id: int):
     
     shared.get_bci_processes_states().pop(process_id)
 
+    collection = db.session.query(BCICollection).get(subprocess_dict['collection_id'])
+    collection.collection_end_time = datetime.utcnow()
+    db.session.commit()
```

Comment:
it would be better to have these three lines before we delete this object at line 153. Also, have you tested that `db.session.commit()` actually writes the datetime correctly? Everything else looks good to me. Make sure to create the database migrations (e.g. `flask db migrate, etc.`

Reply:
Yup tested it and works correctly. I did the migration, although I'm not suppose to add anything from that in the commit right?

Reply:
Tested it and it works. For the migration, I just ran the migrate command, and made sure the scheme of the dev database has been modified.